source/developer/fx-guidelines.rst

```diff
@@ -242,134 +351,113 @@ Input: Hints
 
 -----
 
-Input: Fields
--------------
+Input: Input Patterns
+---------------------
 
-Users should enter information into fields without much thinking. ENTER button on last
-input field should trigger default dialog button. When last input field in a series has
-focus and user hits ENTER it should trigger the default button in the dialog.
+All inputs such as textareas should behave consistently. If the default behaviour is to perform an action on "Enter", then all inputs of that type should be consistent and perform an action on "Enter".
 
 **Example 1:**
+    If pressing "Enter" posts a message in the center channel post input.
 
-    If focus is on the last input field in dialog (“Miller”), hitting ENTER triggers the default dialog button (“Send Invitations”)
-
-    ..  image:: ../images/inputField1.png
-        :alt: Input Field Enter
-        :width: 500 px
+    ..  image:: ../images/inputBehaviour1.png
+        :alt: Center channel post area
 
 **Example 2:**
+    Then pressing "Enter" in the comment thread textarea should also post a comment.
 
-    Having radio buttons for input options.
-    **NOTE**: We should use radio buttons/checkboxes for input options rather than custom bootstrap on/off switches.
+    ..  image:: ../images/inputBehaviour2.png
+        :alt: Comment thread textarea
 
-    ..  image:: ../images/inputField2.png
-        :alt: Radio Buttons
+**Example 3:**
+    And other textareas or inputs should also perform their primary action when "Enter" is pressed, here's an example of the "Edit Header" modal.
+
+    ..  image:: ../images/inputBehaviour3.png
+        :alt: Edit header modal
         :width: 500 px
 
 -----
 
-Display: Button Placement
--------------------------
-
-**Dialog BOTTOM RIGHT BUTTONS should be either “Close”, or “Cancel” and [ACTION_BUTTON].**
-
-If there’s one button on the bottom right, it should be “Close”, if there are two, the one on the left should be “Cancel” and the one on the right should be an [ACTION_BUTTON], like “Save” or “Send Invitations”.
+Input: Validation
+-----------------
 
-**Example:**
+**Feedback on action:**
 
-    Correct:
-        Single button at the bottom right should be “Close”.
+The user should be notified about the action he performed along with any implications associated with it.
```

Comment:
avoid use of "he". Propose "User should be notified about the action they perform along with any implications associated with them."